sdk/servicebus/service-bus/test/internal/streamingReceiver.spec.ts

```diff
@@ -10,10 +10,211 @@ import { ReceiveOptions } from "../../src/core/messageReceiver";
+
+      assert.equal(
+        streamingReceiver["_receiver"]!.credit,
+        101,
+        "Credits are added when receive() is called"
+      );
+
+      assert.isTrue(
+        streamingReceiver.isReceivingMessages,
+        "receive() sets the isReceivingMessages flag immediately to avoid race conditions"
+      );
+
+      assert.equal(streamingReceiver["_receiver"]?.credit, 101);
+
+      // now we'll stop the streaming receiver and then start it back up again.
+      await streamingReceiver.stopReceivingMessages();
+      assert.isFalse(
+        streamingReceiver.isReceivingMessages,
+        "We've stopped receiving messages explicitly"
+      );
+
+      assert.equal(
+        streamingReceiver["_receiver"]?.credit,
+        0,
+        "All receiver credits have been drained"
+      ); // ie, receiver drained
+
+      await streamingReceiver.init(false);
+      assert.isTrue(
+        streamingReceiver.isReceivingMessages,
+        "we've initialized the streaming receiver again so we're ready to receive again"
+      );
+
+      streamingReceiver.subscribe(
+        async (_msg) => {},
+        async (_err) => {}
+      );
+
+      assert.equal(
+        streamingReceiver["_receiver"]?.credit,
+        101,
+        "subscribe has started again, and is revitalized with 101 credits."
+      );
+    });
+
+    it("isReceivingMessages set to false by close()'ing", async () => {
+      const streamingReceiver = new StreamingReceiver(createClientEntityContextForTests());
+      await streamingReceiver.init(true);
+
+      streamingReceiver.subscribe(
+        async (_msg) => {},
+        async (_err) => {}
+      );
+
+      assert.isTrue(streamingReceiver.isReceivingMessages);
+
+      await streamingReceiver.close();
+      assert.isFalse(streamingReceiver.isReceivingMessages);
+    });
+
+    it("isReceivingMessages set to false by calling stopReceivingMessages()", async () => {
+      const streamingReceiver = new StreamingReceiver(createClientEntityContextForTests());
+      await streamingReceiver.init(true);
+
+      streamingReceiver.subscribe(
+        async (_msg) => {},
+        async (_err) => {}
+      );
+
+      assert.isTrue(streamingReceiver.isReceivingMessages);
+
+      await streamingReceiver.stopReceivingMessages();
+      assert.isFalse(streamingReceiver.isReceivingMessages);
+    });
+
+    it("isReceivingMessages set to false by calling detach and init fails", async () => {
+      const streamingReceiver = new StreamingReceiver(createClientEntityContextForTests());
+      await streamingReceiver.init(true);
+
+      streamingReceiver.subscribe(
+        async (_msg) => {},
+        async (_err) => {}
+      );
+
+      assert.isTrue(streamingReceiver.isReceivingMessages);
+
+      streamingReceiver["_init"] = () => {
+        throw new Error("Will never succeed");
+      };
+
+      await streamingReceiver.onDetached(new Error("let's detach"));
+      assert.isFalse(streamingReceiver.isReceivingMessages);
```

Comment:
Basically, this assertion is true because the receiver is set to undefined, right?

Reply:
Correct - the underlying check checks for the receiver not being null and if it's open and the receiver gets removed when close() is called (which happens in onDetach).

We're actually forwarding this call down to the receiverHelper at this point but the check is still the same.